proxy/src/main/java/com/velocitypowered/proxy/command/VelocityCommandManager.java

```diff
@@ -58,6 +58,7 @@ public boolean execute(@NonNull CommandSource source, @NonNull String cmdLine) {
       }
 
       command.execute(source, actualArgs);
+      command.execute(source, split[0], actualArgs);
```

Comment:
Calling the same method with different params might be confusing for API callers. As @astei said on Discord, you already have the command label when you register the command, so I don't fully get the point of this. What's your proposed use case?

Reply:
To allow multiple commands being sent to the same listener to be sorted from there. A more specific example is having a single class route different commands to the same place without having all of them be subcommands of or different classes. IE /ban and /tempban. The only different between them is that tempban needs to look for one more argument of time and without being able to find out the label of the command being run, I would be required to create two almost identical listeners.

Perhaps a better solution would be to have a single execute which gives the label and allow people to ignore it (as done in the Bukkit/Spigot API) or perhaps rename the execute with a label to something more identifiable.

Reply:
Have you thought about creating a `BanCommand` abstract class/interface that subclasses could extend i.e. `TempBanCommand` has to implement the method `ban(Player admin, String hacker, String[] args)` where it could grab the ban duration, reason, etc.

Reply:
Yeah, though I am creating a set of my commands which are unrelated to velocity (to allow for me to create implementations in more apis easily) so i am sending commands through a single class to route them to the common version of the command and I was hoping to avoid having two different sets of each command. To your point though, it would absolutely work it is another set of classes that I was hoping to avoid writing.

Reply:
You can use one class but different objects. Like
```java
public class LabeledCommand implements Command {

    private final ProxyServer server;
    private final String label;
    public  LabeledCommand(ProxyServer server, String label) {
        this.server = server;
        this.label = label;
    }

    @Override
    public void execute(@NonNull CommandSource source, String[] args) {
       if (label.equals("ban") {
         //
       } else if ...{
       //
       } else {
       }
    }
}
```
```java
        server.getCommandManager().register(new  LabeledCommand(server, "ban"), "ban");
        server.getCommandManager().register(new  LabeledCommand(server, "tempban"), "tempban");
        server.getCommandManager().register(new  LabeledCommand(server, "kick"), "kick");
```

Reply:
I suppose the above is a valid solution, though that requires many instances of it would I would prefer to avoid.